### DISCUSSION TITLE
MONGODB DRIVER

### DISCUSSION THREAD

**Discussion by HamedBahram** [nextauthjs/next-auth#2368] [Ideas]
Using Mongodb as my database, I encountered the following error, which is relating to defining write concerns and the deprecation of the used methods in next-auth:

2021-07-13T16:20:32.203Z	e9c48700-94e8-49b4-bbfd-3a4863d427f3	ERROR	(node:8) [MONGODB DRIVER] Warning: Top-level use of w, wtimeout, j, and fsync is deprecated. Use writeConcern instead.

Here is the recommended way of setting write concerns from Mongodb:

**Write concern can include the following fields:**

writeConcern : { w: 'majority', j: true, wtimeout: 1000 }

- the w option to request acknowledgment that the write operation has propagated to a specified number of mongod instances or to mongod instances with specified tags.
- the j option to request acknowledgment that the write operation has been written to the on-disk journal, and
- the wtimeout option to specify a time limit to prevent write operations from blocking indefinitely.

How can this be fixed?

**Comment by balazsorban44** [comment]
which adapter are you using?

I would check our adapters repository https://github.com/nextauthjs/adapters

**Comment by HamedBahram** [comment]
I'm using typeorm.mongodb

Kind Regards,
*Hamed Bahram*

**Reply by balazsorban44** [reply]
https://github.com/nextauthjs/adapters/tree/main/packages/typeorm-legacy here is the source code of that adapter. Can you spot the problem? If not, how could we verify that it is a problem with `next-auth` and not an integration issue between `typeorm` and MongoDB?

**Comment by HamedBahram** [comment]
I didn't spot anything in next-auth source code! It is probably the typeorm
integration with MongoDB.

Here is what I found in typeorm repo
<https://github.com/typeorm/typeorm/blob/master/src/driver/mongodb/MongoConnectionOptions.ts>
setting
connection option in the old way, which will be deprecated.

  The write concern:

   -     readonly w?: string | number
   -     readonly wtimeout?: number
   -     readonly j?: boolean


the correct way is to specify a writeConcern property as an object
containing the above values like so:

writeConcert: { w: 'majority', wtimeout: 1000, j: false }


Kind Regards,
*Hamed Bahram*

**Reply by balazsorban44** [reply]
Then I suggest that you open an issue on their repository, there is nothing for us to do, unfortunately.
